### PR TITLE
Number of arguments section.

### PR DESCRIPTION
**Pull request by neonphlux**
I might be wrong, but ideally you would want one argument (Unary function). Zero arguments implies that it isn't receiving the data in a pure way. And a function that isn't honest in its signature should be avoided.

BTW, thanks for creating this repo, it's very helpful.

### CODE REVIEW

**Comment by ryanmcdermott** [comment]
@neonphlux You've got a very good point. I hadn't thought of it that way! If your function needs to take data and it has zero arguments, some naughty stuff will have to happen, like grabbing global variables. 

I've been trying, and only somewhat successfully, to stay true to the book. On page 71, Martin mentions that "The ideal number of arguments for a function is zero (niladic)" I'm tempted to change it to your insight, I'll let others chime in and see what they think!

I'm glad you found this repo helpful!

**Comment by rkichenama** [comment]
I assumed that Martin was sarcastically joking in the "don't use functions" way; the ideal function is one that does not mutate values given, return expected value(s), and performs as little work as possible.

**Comment by mynameistechno** [comment]
Good point by @neonphlux. In general the book seems to focus on OOP, and I assume a zero arg function is equivalent to a method on an instance object? I.e. the data operated by the function is already a part of the object.

File: README.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ makes testing your function easier. Having more than three leads to a
 combinatorial explosion where you have to test tons of different cases with
 each separate argument.
 
-Zero arguments is the ideal case. One or two arguments is ok, and three should
+One argument is the ideal case. Two arguments is ok, and three should
```
Comment (kwelch) [inline README.md]: Wouldn't it be  `Two arguments are ok, `?

Comment (neonphlux) [inline README.md]: Clumsy me, sorry. 

Fixed.

**Comment by ryanmcdermott** [comment]
Good thoughts here. A function could in theory be "side-effectless" while having zero arguments (such as in case of a function that constructs a very large object and returns it to the caller). In practice though, you all are right to suggest that you're usually going to be providing arguments. Can we have our cake and eat it to here? By that I mean say "Zero or one arguments are ideal" :)

**Comment by ryanmcdermott** [comment]
Closing in favor of having our cake and eating it too 🍰

**Pull request closed without merging**